app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/newdevicenotice/NewDeviceNoticeEmailAccessViewModel.kt

```diff
@@ -0,0 +1,84 @@
+package com.x8bit.bitwarden.ui.auth.feature.newdevicenotice
+
+import android.os.Parcelable
+import androidx.lifecycle.SavedStateHandle
+import com.x8bit.bitwarden.ui.auth.feature.newdevicenotice.NewDeviceNoticeEmailAccessAction.ContinueClick
+import com.x8bit.bitwarden.ui.auth.feature.newdevicenotice.NewDeviceNoticeEmailAccessAction.EmailAccessToggle
+import com.x8bit.bitwarden.ui.auth.feature.newdevicenotice.NewDeviceNoticeEmailAccessEvent.NavigateToTwoFactorOptions
+import com.x8bit.bitwarden.ui.platform.base.BaseViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.update
+import kotlinx.parcelize.Parcelize
+import javax.inject.Inject
+
+private const val KEY_STATE = "state"
+
+/**
+ * Manages application state for the new device notice email access screen.
+ */
+@Suppress("TooManyFunctions")
+@HiltViewModel
+class NewDeviceNoticeEmailAccessViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+) : BaseViewModel<
+    NewDeviceNoticeEmailAccessState,
+    NewDeviceNoticeEmailAccessEvent,
+    NewDeviceNoticeEmailAccessAction,
+    >(
+    initialState = savedStateHandle[KEY_STATE]
+        ?: NewDeviceNoticeEmailAccessState(
+            email = NewDeviceNoticeEmailAccessArgs(savedStateHandle).emailAddress,
+            isEmailAccessEnabled = false,
+        ),
+) {
+    override fun handleAction(action: NewDeviceNoticeEmailAccessAction) {
+        when (action) {
+            ContinueClick -> handleContinueClick()
+            is EmailAccessToggle -> handleEmailAccessToggle(action)
+        }
+    }
+
+    private fun handleContinueClick() {
+        sendEvent(NavigateToTwoFactorOptions)
+    }
+
+    private fun handleEmailAccessToggle(action: EmailAccessToggle) {
+        mutableStateFlow.update {
+            it.copy(isEmailAccessEnabled = action.newState)
```

Comment:
What is the intended purpose of the switch?

Do we not send the code if they do not have access to their email?

Reply:
If the user has access to the email we can dismiss this flow.